--- a/src/usr/local/www/diag_arp.php	2024-09-04 18:42:15
+++ b/src/usr/local/www/diag_arp.php	2024-09-04 18:45:28
@@ -57,6 +57,10 @@
 $arp_table = prepare_ARP_table();

 $pgtitle = [gettext('Diagnostics'), gettext('ARP Table')];
+
+usort($arp_table, function($a, $b) {
+    return ip2long($a['ip-address']) <=> ip2long($b['ip-address']);
+});

 include('head.inc');

@@ -141,7 +145,7 @@
 <?php
 	foreach ($arp_table as $entry): ?>
 				<tr>
-					<td><?=$entry['interface']?></td>
+					<td><?=htmlspecialchars($entry['interface'])?></td>
 					<td><?=$entry['ip-address']?></td>
-					<td><?=$entry['mac-address']?></td>
+					<td><?=htmlspecialchars(strtoupper($entry['mac-address']))?></td>
 					<td><?=$entry['dnsresolve']?></td>
 					<td><?=$entry['expires']?></td>
 					<td><?=$entry['type']?></td>
@@ -151,7 +155,7 @@
 						<a class="fa fa-trash" title="<?=gettext('Delete ARP cache entry')?>" href="diag_arp.php?deleteentry=<?=$entry['ip-address']?>" usepost></a>
 					</td>
 				</tr>
-<?php endforeach; ?>
+<?php endforeach; ?></br>
 				</tbody>
 			</table>
 		</div>
