Espressomatic @UnraidForums - 4 Sep 2024

--- a/src/usr/local/www/status_dhcp_leases.php		2024-09-04 18:42:15
+++ b/src/usr/local/www/status_dhcp_leases.php		2024-09-04 18:45:28
@@ -131,6 +131,10 @@
 $static_string = gettext("static");
 $dynamic_string = gettext("dynamic");
 
+ usort($leases['lease'], function($a, $b) {
+    return strcmp($a['ip'], $b['ip']);
+});
+
 if ($_REQUEST['order']) {
 	usort($leases['lease'], function($a, $b) {
 		return strcmp($a[$_REQUEST['order']], $b[$_REQUEST['order']]);
@@ -292,7 +294,7 @@
 		}
 	}
 
-	$mac = $data['mac'];
+	$mac = strtoupper($data['mac']);
 	$mac_hi = strtoupper($mac[0] . $mac[1] . $mac[3] . $mac[4] . $mac[6] . $mac[7]);
 ?>
 				<tr>
@@ -301,7 +303,7 @@
 <?php if ($data['online'] === $online_string): ?>
 						<i class="fa fa-arrow-up text-success online" title="<?=htmlspecialchars($data['online'])?>"></i>
 <?php else: ?>
-						<i class="fa fa-arrow-down online" title="<?=htmlspecialchars($data['online'])?>"></i>
+						<i class="fa"></i>
 <?php endif; ?>
 					</td>
 					<td><?=htmlspecialchars($data['ip'])?></td>
@@ -312,7 +314,7 @@
 <?php endif; ?>
 						<?=htmlspecialchars($mac)?>
 
-						<?php if (isset($mac_man[$mac_hi])):?>
+						<?php if (isset($mac_man[$mac_hi])):?></br>
 							(<?=htmlspecialchars($mac_man[$mac_hi])?>)
 						<?php endif; ?>
 					</td>
@@ -322,8 +324,8 @@
 						<td><?=htmlspecialchars($data['starts'])?></td>
 						<td><?=htmlspecialchars($data['ends'])?></td>
 					<? else: ?>
-						<td><?=gettext("n/a")?></td>
-						<td><?=gettext("n/a")?></td>
+						<td><?=gettext(" ")?></td>
+						<td><?=gettext(" ")?></td>
 					<? endif; ?>
 					<td>
 <?php if ($data['type'] == $dynamic_string): ?>
